src/main/java/com/example/tweet/controllers/TopController.java

```diff
@@ -71,33 +93,63 @@ public String InsertTweet(HttpServletRequest request, RedirectAttributes redirec
         return "redirect:/";
     }
 
-    @PostMapping("/delete")
-    public String deleteTweet(HttpServletRequest request, RedirectAttributes redirectAttributes) {
-        int id = Integer.parseInt(request.getParameter("id"));
+    @PostMapping("/delete{id}")
+    public String deleteTweet(
+            @RequestParam("id") int tweetId,
+            Model model,
+            RedirectAttributes redirectAttributes) {
         // 削除ボタン押下されたTweetのIDを利用して削除
-        int resultNum = tweetService.deleteTweet(id);
-        System.out.println(resultNum + "件のTweetを削除");
-
-        return "redirect:/";
+        try {
+            tweetService.deleteTweet(tweetId);
+        } catch (NumberOfAffectedRecordError numberEx) {
+            handleException(model, numberEx.getMessage(), numberEx, PATH_ERROR);
+        } catch (DataAccessException dataEx) {
+            handleException(model, "DB操作に失敗しました", dataEx, PATH_ERROR);
+        }
+        redirectAttributes.addFlashAttribute("message", "ツイートを削除しました");
+        return "redirect:/top";
     }
 
     @GetMapping("/edit{id}")
-    public String showEditPage(HttpServletRequest request, Model model) {
-        int tweetId = Integer.parseInt(request.getParameter("id"));
-        // 編集ボタン押下されたTweetのIDを利用して削除
-        Tweet editTargetTweet = tweetService.selectTweetById(tweetId);
+    public String showEditPage(
+            @RequestParam("id") int tweetId,
+            Model model) {
+        // 編集ボタン押下されたTweetのIDを利用してTweet取得
+        try {
+            Tweet editTargetTweet = tweetService.selectTweetById(tweetId);
+            model.addAttribute("tweet", editTargetTweet);
 
-        model.addAttribute("tweet", editTargetTweet);
+        } catch (DataAccessException dataEx) {
+            handleException(model, "DB操作に失敗しました", dataEx, PATH_ERROR);
+        }
         return "/edit.html";
     }
 
     @PostMapping("/edit")
-    public String editTweet(HttpServletRequest request, RedirectAttributes redirectAttributes) {
-        int tweetId = Integer.parseInt(request.getParameter("id"));
-        String content = (String) request.getParameter("content");
-        String updatedMessage = tweetService.updateTweetById(tweetId, content);
+    public String editTweet(
+            @Validated TweetForm tweetform,
+            @RequestParam("id") int tweetId,
+            Model model,
+            RedirectAttributes redirectAttributes) {
+        try {
+            tweetService.updateTweetById(tweetId, tweetform.getContent());
+        } catch (NumberOfAffectedRecordError numberEx) {
```

Comment:
```suggestion
        } catch (NumberOfAffectedRecordError ex) {
```
慣例的に
`e`, `ex`, `exception`あたりが多いですね。
`numberEx`だと数値例外というインスタンスで全く別のものが起因しているのかと勘ぐってしまいそうです。（実務では）

Reply:
インスタンス名をexに変更しました。
46a451682f3d24f029eac3d2f77cb234e69972cd